server/src/test/java/org/opensearch/search/SearchServiceTests.java

```diff
@@ -823,6 +824,118 @@ public Scroll scroll() {
         }
     }
 
+    public void testQuerySearchResultTookTime() throws Exception {
+        // I wasn't able to introduce a delay in these tests as everything between creation and usage of the QuerySearchResult object
+        // happen in a single line - we would have to modify QueryPhase.execute() to take a delay parameter
+        // However this was tested manually
+        createIndex("index");
+        final SearchService service = getInstanceFromNode(SearchService.class);
+        final IndicesService indicesService = getInstanceFromNode(IndicesService.class);
+        final IndexService indexService = indicesService.indexServiceSafe(resolveIndex("index"));
+        final IndexShard indexShard = indexService.getShard(0);
+        SearchRequest searchRequest = new SearchRequest().allowPartialSearchResults(true);
+        searchRequest.source(new SearchSourceBuilder().query(new MatchAllQueryBuilder()));
+
+        ShardSearchRequest request = new ShardSearchRequest(
+            OriginalIndices.NONE,
+            searchRequest,
+            indexShard.shardId(),
+            2, // must have >1 shards for executeQueryPhase to return the QuerySearchResult
+            new AliasFilter(null, Strings.EMPTY_ARRAY),
+            1.0f,
+            -1,
+            null,
+            null
+        );
+
+        SearchShardTask task = new SearchShardTask(123L, "", "", "", null, Collections.emptyMap());
+        service.executeQueryPhase(request, randomBoolean(), task, new ActionListener<SearchPhaseResult>() {
+            @Override
+            public void onResponse(SearchPhaseResult searchPhaseResult) {
+                assertEquals(QuerySearchResult.class, searchPhaseResult.getClass()); // 2+ shards -> QuerySearchResult returned
+                QuerySearchResult qsr = (QuerySearchResult) searchPhaseResult;
+                assertTrue(qsr.getTookTimeNanos() > 0); // Above zero means it's been set at some point
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                throw new AssertionError(e);
+            }
+        });
+    }
+
+    public void testQuerySearchResultTookTimeCacheableRequest() throws Exception {
```

Comment:
For now, we don't need write for cache as such. That will come later when we verify from our own cache tests. This took time related UTs should only be testing whether it is properly setting the value.

Reply:
Ok, I'll remove these tests. We can always reuse the logic if we want them later